test/ui/selenium.go

```diff
@@ -0,0 +1,109 @@
+package ui
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	"image/png"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tebeka/selenium"
+)
+
+//FindElementBy look for a web element by a given selector and returs it back when found.
```

Comment:
One suggestion - I have noticed that there are intermittent timing issues related to the appearance of the UI elements in the Developer Perspective in the UI - I assume that this is due to the DOM being updated. 

It would be a good idea for the test to be robust enough to avoid 'element not fund' errors. I'd suggest making this change:

```
// MaxCount = Max count for loop used to discover UI elements
const MaxCount = 10

//FindElementBy look for a web element by a given selector and returs it back when found.
func FindElementBy(t *testing.T, wd selenium.WebDriver, by string, selector string) selenium.WebElement {

	var elems []selenium.WebElement
	elems, err := wd.FindElements(by, selector)

	// To avoid a problem where the element is yet not present
	require.True(t, len(elems) > 0)
	counter := 0
	for {
		if len(elems) > 0 || counter > MaxCount {
			break
		}
		counter++
		time.Sleep(100 * time.Millisecond)
	}

	elem, err := wd.FindElement(by, selector)
	require.NoError(t, err, fmt.Sprintf("Find element by %s=%s", by, selector))
	return elem
}
```

Reply:
@ldimaggi Thanks for that idea. I implemented it in https://github.com/redhat-developer/devconsole-operator/pull/206/commits/83a06e5d28b9c33cad3f0f7ae19b4df9f85f383f